client/helpers/scenes.ts

```diff
@@ -187,199 +179,319 @@ export async function createSceneAndSceneItems({
   onMsg = console.log,
   onProgress,
   parallelism,
-  polling = { intervalMs: PollIntervalMs, maxAttempts: MaxAttempts },
+  polling = defaultPolling,
   returnQueued = false,
   verbose,
 }: CreateSceneAndSceneItemsReq): Promise<CreateSceneAndSceneItemsRes> {
+  const limit = pLimit(Math.min(parallelism, 100));
+  const startTime = hrtime.bigint();
+  if (verbose) onMsg(`Creating scene...`);
   const scene = (
     await client.scenes.createScene({ createSceneRequest: createSceneReq() })
   ).data;
   const sceneId = scene.data.id;
-  const createRes = await createSceneItems({
-    client,
-    createSceneItemReqs,
-    failFast,
-    onProgress,
-    parallelism,
-    sceneId,
+  if (verbose) onMsg(`Scene ID: ${sceneId}`);
+  if (verbose) onMsg(`Creating scene items...`);
+  let itemCount = 0;
+  let createFailed = false;
+  let sceneResult;
+
+  const reqMap: Map<string, CreateSceneItemRequest[]> = new Map();
+  // create parent map and set ordinals based on request order
+  createSceneItemReqs.forEach((req) => {
+    const reqParent =
+      req.data.attributes.parent ??
+      req.data.relationships.parent?.data.id ??
+      '';
+    if (!reqMap.has(reqParent)) {
+      reqMap.set(reqParent, []);
+    }
+    const siblings = reqMap.get(reqParent);
+    if (req.data.attributes.ordinal == null) {
+      req.data.attributes.ordinal = siblings?.length;
+    }
+    siblings?.push(req);
   });
-  const { a: queuedItems, b: errors } = partition(
-    createRes.queuedSceneItems,
-    (i) => isQueuedJob(i.res)
-  );
-  const queued = returnQueued ? createRes.queuedSceneItems : [];
-  if (queuedItems.length === 0 || errors.length === createRes.leaves) {
-    return { errors, queued, scene };
+
+  // sort all scene item requests into depth sorted array of arrays
+  const depthSortedItems: CreateSceneItemRequest[][] = [];
+  // fetch list of scene items with no parent (root items)
+  let nextChildren: CreateSceneItemRequest[] = reqMap.get('') || [];
```

Comment:
Is `''` the representation of the root?

Will this implementation support multiple root items in the scene? Or only one?